File: VIS.py
fileWithPlayers = [
    {
        "playerName": "Bruninho",
        "number": 1
    },
    {
        "playerName": "Lucao",
        "number": 12
    },
    {
        "playerName": "Giba",
        "number": 10
    }
]

class Team:
    def __init__(self, team_name, players):
        self.name = team_name
        self.players = players

    def print_properties(self):
        print("Name: ", self.name)
        print("Players: ", self.players)

# class Athelet:
class Player:
    def __init__(self, player_name, number):
        self.name = player_name
        self.number = number

    def print_properties(self):
        print("I am ", self.name)
        print("My number is ", self.number)

team = Team("Sada", ["Joao", "Jose", "Josis"])
team.print_properties()


print("Hello, Welcome to Volleyball Information System.")



player = Player("Joao", 1)
player.print_properties()

def para_cada(objeto_a_serMapeado):
    name = objeto_a_serMapeado["playerName"]
    number = objeto_a_serMapeado["number"]
    return Player(name, number)

lista_jogadores_sada = map(para_cada, fileWithPlayers)

print(lista_jogadores_sada)

File: VIS.js
const Team = require( "./Team" ).Team
const Player = require( "./Player" ).Player
const Input = require( "./Skills" ).Input
const fileWithPlayers = require( './file' )
const prompt = require('prompt-sync')({sigint: true});

// criar function para_cada retorne novo Player
function para_cada ( objecto_a_serMapeado ) {
    return new Player( objecto_a_serMapeado.playerName, objecto_a_serMapeado.number )
}

const lista_jogadores_sada = fileWithPlayers.map( para_cada )
const team_sada = new Team( "Sada", lista_jogadores_sada )
const player = team_sada.players.find( playerName => playerName.number === 10 )

new Input( team_sada, "serve", player, "+" )
new Input( team_sada, "serve", player, "+" )
new Input( team_sada, "serve", player, "+" )
new Input( team_sada, "serve", player, "+" )



console.log( player )


 
// Random number from 1 - 10
const numberToGuess = Math.floor(Math.random() * 10) + 1;
// This variable is used to determine if the app should continue prompting the user for input
let foundCorrectNumber = false;
 
while (true) {
  // Get user input
  let skill = prompt('Skill ');
  let number = prompt('Number ')
  let result = prompt('Result ')
  // Convert the string input to a number
  const player = team_sada.players.find( playerName => playerName.number === parseInt(number) )
  new Input( team_sada, skill, player, result )
 
    console.log(player)
}
